dependencies/code_builder/buildspec.yml

```diff
@@ -8,38 +8,61 @@ phases:
         PULL_FROM_DOCKER=("grafana/grafana" "grafana/loki" "grafana/promtail" "grafana/fluent-bit-plugin-loki" "nginx" "juspaydotin/hyperswitch-router" "juspaydotin/hyperswitch-producer" "juspaydotin/hyperswitch-consumer" "juspaydotin/hyperswitch-control-center" "juspaydotin/hyperswitch-web" "bitnami/metrics-server")
         PULL_FROM_AWS=("eks/aws-load-balancer-controller" "ebs-csi-driver/aws-ebs-csi-driver" "eks-distro/kubernetes-csi/external-provisioner" "eks-distro/kubernetes-csi/external-attacher" "eks-distro/kubernetes-csi/external-snapshotter/csi-snapshotter" "eks-distro/kubernetes-csi/livenessprobe" "eks-distro/kubernetes-csi/external-resizer" "eks-distro/kubernetes-csi/node-driver-registrar" "ebs-csi-driver/volume-modifier-for-k8s")
 
-        for IMAGE in "${PULL_FROM_DOCKER[@]}"; do
-          aws ecr create-repository --repository-name $IMAGE --image-scanning-configuration scanOnPush=true
-        done
+        repository_exists() {
+          local repo_name="$1"
+          aws ecr describe-repositories --repository-names "$repo_name" >/dev/null 2>&1
+        }
+
+        image_exists() {
+          local repo_name="$1"
+          local image_tag="$2"
+          aws ecr describe-images --repository-name "$repo_name" --image-ids imageTag="$image_tag" >/dev/null 2>&1
+        }
+
+        delete_image() {
+          local repo_name="$1"
+          local image_tag="$2"
+          aws ecr batch-delete-image --repository-name "$repo_name" --image-ids imageTag="$image_tag"
+        }
```

Comment:
We have tag immutability disabled for the repositories that are created, So, we can upload the same tag multiple times into the repository without any error! Deletion isn't required.